### PR TITLE
Flutter-specific dartanalyzer options.

### PR DESCRIPTION
**Pull request by isoos**
We can fine-tune how to synchronize the linter rules between the two, I've just copied the flutter's options file as it was to unblock supermixin.

### CODE REVIEW

File: lib/src/analysis_options.dart [hunk 1]
```diff
@@ -13,8 +16,80 @@ linter:
...
+  strong-mode: true
+  errors:
+    # treat missing required parameters as a warning (not a hint)
+    missing_required_param: warning
+    # treat missing returns as a warning (not a hint)
+    missing_return: warning
+    # allow having TODOs in the code
+    todo: ignore
+
+linter:
```
Comment (kevmoo) [inline lib/src/analysis_options.dart]: Let's remove these. return types on setters, use isNotEmpty, etc have no affect on runtime.

Let's use one set of lints and just turn on the language, strong-mode, and error tweaks

Comment (isoos) [reply lib/src/analysis_options.dart]: Like this? or do you want to have the error tweaks in the default options too?

Comment (kevmoo) [reply lib/src/analysis_options.dart]: Yup!

We could even share the lints in a variable...but not a biggy.

Comment (isoos) [reply lib/src/analysis_options.dart]: yeah. the benefit of this version: easy to copy-paste. Can do the variable thing if you prefer that.

**Comment by kevmoo** [comment]
What version should this go in? 0.8.1?

Should flag as -dev, etc...unless we want to roll this out with the Pub-host changes, too...

**Comment by isoos** [comment]
> What version should this go in? 0.8.1?

I omitted that because I didn't want to have a conflict with the other PR. I'll change it to 0.8.1 there and we can append the changelog of this PR.

File: lib/src/analysis_options.dart (hunk 1 above)
Comment (kevmoo) [reply lib/src/analysis_options.dart]: No biggy

**Comment by kevmoo** [comment]
> I omitted that because I didn't want to have a conflict with the other PR. I'll change it to 0.8.1 there and we can append the changelog of this PR.

Okay. I really like using -dev for pushed that aren't designed to be published. Not a biggy, though.

**Pull request merged**